### PR TITLE
#167165071 get all users

### PR DESCRIPTION
**Pull request by PelumiAlesh**
### What does this PR do?
Build endpoint to get all users
#### Description of Task to be completed?
Get Users
build an endpoint for getting all users
write test for the endpoint
document the endpoint with swagger
#### How should this be manually tested?
make a request to `api/v1/users`
#### What are the relevant pivotal tracker stories?
#167165071 
#### Screenshots (if appropriate)
Screenshot 2019-08-08 at 7 15 10 PM

### CODE REVIEW

File: __tests__/getAllUsers.test.js [hunk 1]
```diff
@@ -26,7 +26,19 @@ describe('Testing GET users endpoint', () => {
         done();
       });
   });
-
+  it('should get all articles successfully', (done) => {
```
Comment (mazma1) [inline __tests__/getAllUsers.test.js]: Get all articles??

Comment (PelumiAlesh) [reply __tests__/getAllUsers.test.js]: Fixed

File: controllers/UsersController.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+   * @param {object} res express response object
+   * @param {function} next
+   * @returns {object} returns all user
+   */
+  static async getAllUsers(req, res, next) {
+    try {
+      const allUsers = await models.User.findAll({
+        attributes: ['id', 'firstName', 'lastName', 'email', 'isVerified', 'type', 'emailNotify'],
+        include: [
+          {
```
Comment (Tunji17) [inline controllers/UsersController.js:27]: Nice one you have included the profile

**Review by Tunji17 — APPROVED** [review]

File: __tests__/getAllUsers.test.js [hunk 3]
```diff
@@ -26,7 +26,19 @@ describe('Testing GET users endpoint', () => {
...
+    chai
+      .request(app)
+      .get(url)
+      .set('Authorization', `Bearer ${validToken}`)
+      .end((err, res) => {
+        expect(res.status).to.eql(200);
+        expect(res.body.users).to.have.property('message');
+        expect(res.body.users.message).to.eql('Users fetched successfully');
+        expect(res.body.users).to.be.a('object');
```
Comment (mazma1) [inline __tests__/getAllUsers.test.js:38]: Don't we have at least one specific user you are returning?

**Comment by mazma1** [comment]
Also, update your screenshot so I can see what the response looks like now

File: __tests__/getAllUsers.test.js (hunk 3 above)
Comment (PelumiAlesh) [reply __tests__/getAllUsers.test.js:38]: I don't think there's a way I can ascertain the specific user. can you please add more color @mazma1

Comment (mazma1) [reply __tests__/getAllUsers.test.js:38]: Your response message says `Users fetched successfully`. So which users did you fetch?

Comment (PelumiAlesh) [reply __tests__/getAllUsers.test.js:38]: Oh Okay, I would work on the test now.

**Comment by mazma1** [comment]
From your screenshot, what's the difference between `allUsers[0].id` and `allUsers[0].profile.id`?

Since we have the user's id `allUsers[0].id`, do you necessarily have to include `allUsers[0].profile.userId`?

**Review by VictorMaria — APPROVED** [review]

**Pull request merged**